src/plugins/intel_npu/src/plugin/npuw/weights_bank.cpp

```diff
@@ -70,6 +70,10 @@ ov::Tensor Bank::get(int64_t uid, const std::string& device) {
     // uid may be coming from a 2nd (3rd, ...) model
     // detach the tensor here just in case
     const_cast<LazyTensor&>(iter_device->second.lt).detach();
+    // Also detach from registered tensors
+    auto it_registered = device_bank.registered_tensors.find(iter_device->second.lt);
```

Comment:
Why doesn't this happen on `evaluate_and_allocate()` ?

Reply:
Done